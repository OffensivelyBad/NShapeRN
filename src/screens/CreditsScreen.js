import React from 'react';
import { FlatList, View } from 'react-native';
import { SimpleTableView } from '../components/common/';

const credits = [
  { id: '1', title: 'Shane Rocker', subtitle: 'Development' },
  { id: '2', title: 'Lindsey Roller', subtitle: 'Inspiration' },
  { id: '3', title: 'Jason Worrix', subtitle: 'Design' },
  { id: '4', title: 'Brony Fox', subtitle: 'QA' },
]

class CreditsScreen extends React.PureComponent {

  static navigationOptions = {
    title: 'Credits',
  }

  _keyExtractor = (item) => item.id;

  _renderItem = ({item}) => {
    return (
      <SimpleCell 
        key={item.id}
        item={{...item, hasSwitch: false}}
        backgroundColor='#fff' 
        textColor='#333'
        textMarginLeft={50}
        height={50}
        fontSize={30}
      >
        {item.title}
    </SimpleCell>
    )
  }

  render() {
    return (
      <View style={styles.containerStyle}>
        <SimpleTableView data={credits} onPressItem={() => {}} />
      </View>
    );
  }

}

const styles = {
  containerStyle: {
    flex: 1,
  }
}

export default CreditsScreen;
