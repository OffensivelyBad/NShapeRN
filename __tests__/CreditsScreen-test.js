import React from 'react';
import CreditsScreen from '../src/screens/CreditsScreen';
import renderer from 'react-test-renderer';

test('CreditsScreen renders correctly', () => {
    const tree = renderer.create(<CreditsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});
