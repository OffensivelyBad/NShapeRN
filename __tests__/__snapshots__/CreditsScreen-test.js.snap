// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "id": "1",
            "subtitle": "Development",
            "title": "Shane Rocker",
          },
          Object {
            "id": "2",
            "subtitle": "Inspiration",
            "title": "Lindsey Roller",
          },
          Object {
            "id": "3",
            "subtitle": "Design",
            "title": "Jason Worrix",
          },
          Object {
            "id": "4",
            "subtitle": "QA",
            "title": "Brony Fox",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Shane Rocker
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            >
              Development
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Lindsey Roller
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            >
              Inspiration
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Jason Worrix
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            >
              Design
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Brony Fox
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            >
              QA
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
